system/jlib/jstring.cpp

```diff
@@ -1809,6 +1809,9 @@ int utf8CharLen(const unsigned char *ch, unsigned maxsize)
 
 const char *encodeXML(const char *x, StringBuffer &ret, unsigned flags, unsigned len, bool utf8)
 {
+    // This is the minimum length that will get generated, and potentially saves large number of relocations for large strings
+    if (len != (unsigned)-1)
```

Comment:
Possibly better is 
```
if (len == (unsigned)-1)
    len = strlen(x);
ret.ensureCapacity(len);
```
and avoid the test in the main loop.

Reply:
It would be more readable that way and for the string sizes that matter here the ensureCapacity() likely wins vs the extra strlen?